apps/dashboard/src/hooks/useResolveContractAbi.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from "@tanstack/react-query";
+import type { Abi } from "abitype";
+import type { ThirdwebContract } from "thirdweb";
+import { resolveContractAbi } from "thirdweb/contract";
+import invariant from "tiny-invariant";
+
+export function useResolveContractAbi(contract?: ThirdwebContract) {
+  return useQuery({
+    queryKey: [
+      "full-contract-abi",
+      contract?.address || "",
+      contract?.chain.id || "",
```

Comment:
i would invert chain and address here, you always want the query key to go broad -> narrow, so you can invalidate for example the whole chain, or just a specific contract on a chain.

right now the way you have it if i want to invalidate a whole chain i can't because contract address comes first

Reply:
this is new to me. ty!